Cask was detected as formula with `--only-formulae`

I have a tap providing casks only. Today when I opened https://github.com/muzimuzhi/homebrew-texstudio-beta/pull/16 to update cask `texstudio-beta`, it was detected as formula hence the workflow failed with log
```
Run brew test-bot --only-formulae --root-url=https://ghcr.io/v2/muzimuzhi/homebrew-texstudio-beta
==> Using Homebrew/homebrew-test-bot 09eca62 (Merge pull request #896 from carlocab/remove-debug-output)
==> Using Homebrew/brew [4](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:5).0.10-119-g931327df1 (Merge pull request #15116 from ZhongRuoyu/bump-cask-pr-sha256)
==> Testing muzimuzhi/homebrew-texstudio-beta 2e3a69f (Merge 4aed83a1[5](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:6)d3fecf22d961f7a092d764e72b56838 into da40a90d1b055e6c123d1b926eefd2290a241861):

==> Running FormulaeDetect#detect_formulae!
==> git -C /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/muzimuzhi/homebrew-texstudio-beta fetch origin +refs/heads/master
    url               https://github.com/muzimuzhi/homebrew-texstudio-beta/pull/1[6](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:7)/checks
    tap origin/master da40a90 (texstudio: fix style violations)
    HEAD              2e3a69f (Merge 4aed83a15d3fecf22d961f[7](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:8)a092d764e72b56[8](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:9)38 into da40a[9](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:10)0d1b055e6c[12](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:13)3d1b926eefd2290a241861)
    diff_start_sha1   da40a90d1b055e6c123d1b926eefd2290a241861
    diff_end_sha1     2e3a69fbc6b7a778c4[14](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422#step:8:15)4dbf803957f1e8be2a08

    testing_formulae  muzimuzhi/texstudio-beta/texstudio-beta
    added_formulae    (none)
    modified_formulae muzimuzhi/texstudio-beta/texstudio-beta
    deleted_formulae  (none)
Error: No available formula with the name "muzimuzhi/texstudio-beta/texstudio-beta".
Error: Process completed with exit code 1.
```
In the past, the log of such step looked like
```
==> Running FormulaeDetect#detect_formulae!
==> git -C /usr/local/Homebrew/Library/Taps/muzimuzhi/homebrew-texstudio-beta fetch origin +refs/heads/master
    url               https://github.com/muzimuzhi/homebrew-texstudio-beta/pull/15/checks
    tap origin/master 1bac692 (texstudio-beta 4.5.1beta6)
    HEAD              3ff9554 (Merge 099a069fcdc5bc86eea44cb751d716716[12](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4031951927/jobs/6931540308#step:8:13)d8704 into 1bac692b95acf1a2ab4a8d8aa56ed42[13](https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4031951927/jobs/6931540308#step:8:14)dcc0d63)
    diff_start_sha1   1bac692b95acf1a2ab4a8d8aa56ed4213dcc0d63
    diff_end_sha1     3ff9554480db77dd866728012c66f9e5b86853a6

    testing_formulae  (none)
    added_formulae    (none)
    modified_formulae (none)
    deleted_formulae  (none)
All steps passed!
```

- Today's failed workflow run https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589072932/jobs/8103716422
- A successful workflow run in the past (on Jan 28, 2023) https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4031951927/jobs/6931540308

No luck, see https://github.com/muzimuzhi/homebrew-texstudio-beta/actions/runs/4589702164/jobs/8104777253
```
Run brew test-bot --only-formulae --root-url=https://ghcr.io/v2/muzimuzhi/homebrew-texstudio-beta
==> Using Homebrew/homebrew-test-bot 07b949e (Revert "Formula: "shard" directory.")
[...]
    testing_formulae  muzimuzhi/texstudio-beta/texstudio-beta
    added_formulae    (none)
    modified_formulae muzimuzhi/texstudio-beta/texstudio-beta
    deleted_formulae  (none)
Error: No available formula with the name "muzimuzhi/texstudio-beta/texstudio-beta".
Error: Process completed with exit code 1.
```

--- a/lib/tests/formulae_detect.rb
+++ b/lib/tests/formulae_detect.rb
@@ -191,6 +191,7 @@ module Homebrew
           start_revision, end_revision, "--", path
         ).lines.map do |line|
           file = Pathname.new line.chomp
+          next if tap.cask_file?(file)
           next unless tap.formula_file?(file)
 
           tap.formula_file_to_name(file)

